subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -4008,6 +4014,10 @@ int lwm2m_engine_context_close(struct lwm2m_ctx *client_ctx)
 
 void lwm2m_engine_context_init(struct lwm2m_ctx *client_ctx)
 {
+	coap_pendings_clear(client_ctx->pendings,
```

Comment:
I've meant those as well, I'd see them in `lwm2m_engine_context_close`, no point splitting cleanup operations.